src/Microsoft.Dnx.Tooling/Publish/PublishProject.cs

```diff
@@ -107,6 +107,17 @@ private bool EmitNupkg(PublishRoot root)
                 root.Operations.Delete(TargetPath);
             }
 
+            // If this is a wrapper project, we need to generate a lock file before building it
+            if (IsWrappingAssembly())
+            {
+                var success = Restore(root, publishProject: this, restoreDirectory: project.ProjectDirectory)
```

Comment:
However, do we really care the "publish without packages installed" scenario? If you see a publish failure, you can restore and retry.